### PR TITLE
Adapt source-mssql to latest Kotlin converted CDK

### PR DESCRIPTION
**Pull request by rodireich**
Make source-mssql build and test with the latest CDK

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/JdbcDatabase.kt [hunk 1]
```diff
@@ -111,7 +111,7 @@ abstract class JdbcDatabase(protected val sourceOperations: JdbcCompatibleSource
      * close the returned stream to release the database connection. Otherwise, there will be a
      * connection leak.
      *
-     * @param statementCreator create a [PreparedStatement] from a [Connection].
+     * @paramstatementCreator create a [PreparedStatement] from a [Connection].
```
Comment (xiaohansong) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/JdbcDatabase.kt:114]: nit: revert this

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/JdbcDatabase.kt [hunk 2]
```diff
@@ -195,10 +195,10 @@ abstract class JdbcDatabase(protected val sourceOperations: JdbcCompatibleSource
     }
 
     @Throws(SQLException::class)
-    fun queryMetadata(sql: String, vararg params: String): ResultSetMetaData {
+    fun queryMetadata(sql: String, vararg params: String): ResultSetMetaData? {
         unsafeQuery(
                 { c: Connection -> getPreparedStatement(sql, params, c) },
-                { obj: ResultSet -> obj.metaData }
+                { obj: ResultSet -> obj.metaData },
```
Comment (xiaohansong) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/JdbcDatabase.kt:201]: do we need this extra comma?

**Review by xiaohansong — APPROVED** [review]
some minor nits need to be cleaned up, otherwise LGTM!

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/JdbcDatabase.kt (hunk 2 above)
Comment (rodireich) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/JdbcDatabase.kt:201]: I think this is something our formatter does

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/JdbcDatabase.kt (hunk 1 above)
Comment (rodireich) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/JdbcDatabase.kt:114]: Formatter did this 🤷‍♂️ 
I don't quite know why

**Pull request merged**